### DISCUSSION TITLE
using GridStat with goofy long pressure levels

### DISCUSSION THREAD

**Discussion by sherrieF** [dtcenter/METplus#1807] [Configuration] [answered]
Hello,
    I would like to use METplus GridStat with ERA5 data.  My forecast input file has goofy pressure levels.  For example
847.7790484268779.  My ERA5 data has pressure levels like 850.  Do I need to interpolate the goofy pressure levels
to the pressure levels that I have in the ERA5 data.  If so I am not sure of how to do the interpolation.  I have tried to
run GridStat with the goofy pressure levels.  I get an error saying the pressure level is not found.
I hope this makes sense,
Thanks,
Sherrie Fredrick

**Comment by j-opatz** [comment]
Hi Sherrie, 

And thank you for your question. It's unfortunate that the files are set up as they are, and without an example in hand I'm reluctant to say if there's a straight path forward or not.

Would you be able to share an example of the non-standard pressure level files? [This](https://github.com/dtcenter/METplus/discussions/954) post will help with steps on how to send us the data.

Hopefully with a chance to test with the data, we can find a MET environment solution for you!

**Comment by sherrieF** [comment]
Hi Julie,

    I tried to send to send my files to ftp -p ftp.rap.ucar.edu, but I
received

***@***.*** 2015122800]$ ftp -p ftp.rap.ucar.edu


ftp: connect: Connection timed out


I am working on Hera.  My Hera directory is
/scratch2/BMC/gsienkf/Sherrie.Fredrick/Laura/forecast/cntrl/.
My

conf file is GridStat.conf in this directory.


I am also working on a solution using the NCL function int2p_n_Wrap.  This
function interpolates from one pressure level

to another.  This will solve my problem.  The only thing is that I will
need to run this on all of my data to produce

a new netcdf file.  This interpolation may need to be done anyway since my
era5 data is on slightly different pressure levels than my forecast data
and I don't know if METplus likes this or not.

Have a great weekend and thanks for your help.

Sherrie

**Reply by j-opatz** [reply]
Hi Sherrie,

I wanted to reach out to let you know we're working on this! I was able to go into Hera and grab a sample file (sfg_2015122800.nc) to look for possible solutions. I'll let you know if we come up with anything.

**Comment by sherrieF** [comment]
Hi Julie.

      Don't worry about it too much.  I am able to interpolate with NCL.
 Also I am not sure that the pfull are pressure levels.  They may be
module levels since they
are so goofy.

Thanks,
Sherrie

**Comment by sherrieF** [comment]
Hi John,

    I contacted the person I received the data from and the vertical levels
are not pressure levels, They are model levels.  Even though the meta data
calls them pressure levels.  This would explain the very goofy values.  So
I can't use METplus to do stats on them I don't believe.

Thanks,
Sherrie

**Comment by j-opatz** [comment, marked as answer]
Hi Sherrie,

John HG and I just finished up a working session on an example data file (sfg_2015122800.nc) that you indicated was giving issues in MET. Unfortunately as you already concluded, there's no way within MET to interpolate from a given level to another, one of the drawbacks of model vs standard pressure levels. So from a 1:1 viewpoint, you'd be unable to compare the model data to a standard pressure level output without some assumptions/accepting uncertainties in the verification values.

However, there are some ways to make the data as it currently resides in the file passable to MET, if you're still in need of it.

MET 10.1 requires the index of the nc dimension to be specified: for example, when trying to pass the file into a simple plot_data_plane call,

`/usr/local/met-10.1.0/bin/plot_data_plane sfg_2015122800.nc ~/plot.ps  'name="ugrd"; level="(0,34,*,*)"; file_type=NETCDF_NCCF;'`

Note that this call requires you to tell MET that the file is NETCDF_NCCF style, as the conventions were not listed inside the netCDF itself. But by specifying the desired time and pfull indicies, an image was successfully created.

MET 11.0.0 beta3 (or develop) supports the @value notation that Howard added. Using the same example as before but with the new notation,

`/d1/projects/MET/MET_regression/develop/NB20220922/MET-develop/bin/plot_data_plane sfg_2015122800.nc ~/plot.ps  'name="ugrd"; level="(0,@780.640153369,*,*)"; file_type=NETCDF_NCCF;'`

This allows a user to call the value directly, instead of trying to figure out what order the levels are indexed in. You may have noticed in the example above that the entire pressure level is not written out. This is because the comparison MET does between what you as the user enter and the pressure field it looks for in the input file only needs a certain level of accuracy, approximately 1.0xe-9 or so. So while the above example would return the level, the following does not:

`/d1/projects/MET/MET_regression/develop/NB20220922/MET-develop/bin/plot_data_plane sfg_2015122800.nc ~/plot.ps  'name="ugrd"; level="(0,@780.64015336,*,*)"; file_type=NETCDF_NCCF;'`

Hopefully this helps a bit, even if the pressure interpolation isn't currently possible in MET.

**Comment by sherrieF** [comment]
Hi John,

    Thanks you so much for your help with this.  Your email helped a lot.
Please consider the discussion closed and again many thanks,
Sherrie
